shared/qt/idian/widgets/Row.cpp

```diff
@@ -0,0 +1,371 @@
+/******************************************************************************
+    Copyright (C) 2023 by Dennis Sädtler <dennis@obsproject.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include <QComboBox>
+#include <QApplication>
+#include <QSizePolicy>
+#include <QSvgRenderer>
+
+#include <Idian/moc_Row.cpp>
+
+namespace idian {
+
+Row::Row(QWidget *parent) : GenericRow(parent)
+{
+	layout = new QGridLayout(this);
+	layout->setVerticalSpacing(0);
+	layout->setContentsMargins(0, 0, 0, 0);
+
+	labelLayout = new QVBoxLayout();
+	labelLayout->setSpacing(0);
+	labelLayout->setContentsMargins(0, 0, 0, 0);
+
+	setFocusPolicy(Qt::StrongFocus);
+	setLayout(layout);
+
+	layout->setColumnMinimumWidth(0, 0);
+	layout->setColumnStretch(0, 0);
+	layout->setColumnStretch(1, 40);
+	layout->setColumnStretch(2, 55);
+
+	nameLabel = new QLabel();
+	nameLabel->setVisible(false);
+	Utils::addClass(nameLabel, "title");
+
+	descriptionLabel = new QLabel();
+	descriptionLabel->setVisible(false);
+	Utils::addClass(descriptionLabel, "description");
+
+	labelLayout->addWidget(nameLabel);
+	labelLayout->addWidget(descriptionLabel);
+
+	layout->addLayout(labelLayout, 0, 1, Qt::AlignLeft);
+}
+
+void Row::setPrefix(QWidget *w, bool auto_connect)
+{
+	setSuffixEnabled(false);
+
+	_prefix = w;
```

Comment:
Don't use underscore prefixes in C++, those might clobber reserved internal names used by the platform library.